node/utils/metrics/metrics.ts

```diff
@@ -2,25 +2,13 @@ import axios from 'axios'
 
 const ANALYTICS_URL = 'https://rc.vtex.com/api/analytics/schemaless-events'
 
-type ImpersonateUserMetric = {
-  kind: 'impersonate-user-graphql-event'
-  description: 'Impersonate User Action - Graphql'
+export interface Metric {
+  readonly account: string
+  readonly kind: string
+  readonly description: string
+  readonly name: 'b2b-suite-buyerorg-data'
 }
 
-type ImpersonateB2BUserMetric = {
-  kind: 'impersonate-b2b-user-graphql-event'
-  description: 'Impersonate B2B User Action - Graphql'
-}
```

Comment:
why did we remove these `Impersonate` types? Don't we have a metric about impersonate user usage? Should we really remove this? I believe you are trying to make the code more generic here, but I don't see the kind/description for the impersonate metric being added anywhere else

Reply:
Yes, you are right! I changed the code to be more flexible because, for each metric, it was needed to put the class in this type.
```Typescript
export type Metric = {
  name: 'b2b-suite-buyerorg-data'
  account: string
} & (
  | ImpersonateUserMetric
  | ImpersonateB2BUserMetric
  | UpdateOrganizationMetric
)
```

I don't know why, but this parameters are declared here and in [the metrics function](https://github.com/vtex-apps/b2b-organizations-graphql/blob/master/node/utils/metrics/impersonate.ts#L66). So I prefere to remove the redundancy refactoring the code.